src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala

```diff
@@ -430,8 +430,13 @@ trait ContextErrors {
       }
 
       //typedNew
-      def IsAbstractError(tree: Tree, sym: Symbol) = {
-        issueNormalTypeError(tree, sym + " is abstract; cannot be instantiated")
+      def IsAbstractError(tree: Tree, sym: Symbol, context: Context) = {
+        issueNormalTypeError(tree, sym + s" is ${if (sym.isTrait) "trait" else "abstract"}; cannot be ${if(context.typingAnnotationCore) "used as an annotation" else "instantiated"}")
```

Comment:
Might as well write `s"$sym is ${ stuff }"`. Avoids `any2stringadd`, right? which actually warns now.